packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -358,26 +358,25 @@ class AndroidDevice extends Device {
     if (!await _checkForSupportedAdbVersion() || !await _checkForSupportedAndroidVersion())
       return LaunchResult.failed();
 
-    final TargetPlatform devicePlatform = await targetPlatform;
-    if (!(devicePlatform == TargetPlatform.android_arm ||
-          devicePlatform == TargetPlatform.android_arm64) &&
-        !(debuggingOptions.buildInfo.isDebug ||
-          debuggingOptions.buildInfo.isDynamic)) {
-      printError('Profile and release builds are only supported on ARM targets.');
-      return LaunchResult.failed();
-    }
-
     BuildInfo buildInfo = debuggingOptions.buildInfo;
-    if (buildInfo.targetPlatform == null && devicePlatform == TargetPlatform.android_arm64)
-      buildInfo = buildInfo.withTargetPlatform(TargetPlatform.android_arm64);
+    if (buildInfo.targetPlatform == null) {
+      buildInfo = buildInfo.withTargetPlatform(await targetPlatform);
+    }
 
     if (!prebuiltApplication || androidSdk.licensesAvailable && androidSdk.latestVersion == null) {
       printTrace('Building APK');
       final FlutterProject project = FlutterProject.current();
-      await buildApk(
-          project: project,
-          target: mainPath,
-          buildInfo: buildInfo,
+      PlatformBuildStep platformStep;
+      try {
+        platformStep = platformBuilders.selectPlatform(buildInfo: buildInfo);
+      } on Exception catch (err) {
```

Comment:
Catch the specific exceptions, or change `selectPlatform` to return an object that distinguishes among the different kinds of errors.